client/src/app/formulaire/formulaire.component.ts

```diff
@@ -29,27 +29,34 @@ export class FormulaireComponent implements OnInit {
 
 
   constructor(private apollo:Apollo, private activateRouter:ActivatedRoute, private router:Router) {
-    //  console.logthis.activateRouter.snapshot.paramMap.get('email')
-    console.log(JSON.stringify(activateRouter.snapshot.paramMap))
+    
    }
 
 
   ngOnInit(): void {
+    const params = this.activateRouter.snapshot.paramMap.get('params');
+    let data;
+    if(params)
+    data = decodeURIComponent(params!).split('&')
+    console.log(data)
     this.sendRequest = new SendRequest();
+    if(data){
+      this.sendRequest.nom = data[0].split('=')[1]
+      this.sendRequest.email = data[1].split('=')[1]
+      this.sendRequest.receverName = data[2].split('=')[1]
+      this.sendRequest.receverEmail = data[3].split('=')[1]
+    }
```

Comment:
Dans les urls il existe déjà un mécanisme pour passer des paramètres, ça s'appel des query params 
ça se trouve de la forme /formulaire/?nom=sdf&email=sdfsdf

Et c'est tellement extraordinaire, c'est que angular est même capable de les récupérer !
https://angular.io/guide/router#accessing-query-parameters-and-fragments

Reply:
Ton serveur doit être capable d'encoder ta chaine pour en faire un query param.